aim/web/ui/src/services/api/api.ts

```diff
@@ -346,13 +348,20 @@ async function checkCredentials<T>(
   if (response.status === 401) {
     if (endpoint === `${ENDPOINTS.AUTH.BASE}/${ENDPOINTS.AUTH.REFRESH}`) {
       removeAuthToken();
+      window.location.assign(`${window.location.origin}/sign-in`);
       return parseResponse<T>(response);
     }
-    // Refresh token
-    const token = await refreshToken().call();
-    if (token) {
-      setAuthToken(token);
-      return refetch();
+    if (localStorage.getItem('refreshing') !== 'true') {
+      localStorage.setItem('refreshing', 'true');
```

Comment:
Isn't there a possibility for race conditions here? Like, reading the 'refreshing' key right before it is set to 'true'?